R/zchunk_LB181.ag_R_C_Y_GLU_irr_mgmt.R

```diff
@@ -52,83 +54,351 @@ module_aglu_LB181.ag_R_C_Y_GLU_irr_mgmt_DISABLED <- function(command, ...) {
+    # Method ONE: calculate a generic weighted lo, observed, and hi yield across all crops
+    L181.YieldLevels_R_C_GLU_irr %>%
+      group_by(GCAM_region_ID, GLU, Irr_Rfd) %>%
+      summarise(HA_ha = sum(HA_ha), wt_yieldmult_lo = sum(wt_yieldmult_lo), wt_yieldmult_hi = sum(wt_yieldmult_hi)) %>%
+      ungroup %>%
+      mutate(yieldmult_lo = wt_yieldmult_lo / HA_ha, yieldmult_hi = wt_yieldmult_hi / HA_ha,
+             # Do not allow the bioenergy "hi" yield multipliers exceed some exogenous threshold
+             yieldmult_hi = pmin(yieldmult_hi, MAX_BIO_MULTI_HI)) ->
+      L181.YieldLevels_R_GLU_irr
+
+    # Method TWO: APPLYING HACK TO SET THE SAME YIELD MULTIPLIERS EVERYWHERE
+    L181.YieldLevels_R_GLU_irr %>%
+      mutate(yieldmult_hi = 1 + MGMT_YIELD_ADJ, yieldmult_lo = 1 - MGMT_YIELD_ADJ) %>%
+      select(-wt_yieldmult_lo, -wt_yieldmult_hi, -HA_ha) ->
+      L181.YieldMult_R_bio_GLU_irr
+
+    # Calculate bioenergy land shares
+    # Again, two methods are developed, and currently adopts method TWO
+    # Method ONE:
+    L181.YieldLevels_R_GLU_irr %>%
+      mutate(landshare_lo = (1 - yieldmult_hi) / (yieldmult_lo - yieldmult_hi), landshare_hi = 1 - landshare_lo) %>%
+      select(GCAM_region_ID, GLU, Irr_Rfd, landshare_lo, landshare_hi) ->
+    L181.LandShare_R_bio_GLU_irr
+
+    # Method TWO: APPLYING HACK TO SET THE SAME LADN SHARE
+    L181.LandShare_R_bio_GLU_irr %>%
+      mutate(landshare_lo = 0.5, landshare_hi = 0.5) ->
+      L181.LandShare_R_bio_GLU_irr
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L181.LC_bm2_R_C_Yh_GLU_irr_level %>%
+      add_title("Cropland cover by GCAM region / commodity / year / GLU / irrigation / mgmt level") %>%
+      add_units("bm2") %>%
+      add_comments("Cropland cover by high and low management levels are currently set at the share of 50% by each.") %>%
       add_legacy_name("L181.LC_bm2_R_C_Yh_GLU_irr_level") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("common/iso_GCAM_regID",
+                     "aglu/LDS/Mueller_yield_levels",
+                     "aglu/Muller_crops",
+                     "aglu/FAO_ag_items_PRODSTAT",
+                     "L151.ag_irrHA_ha_ctry_crop",
+                     "L151.ag_rfdHA_ha_ctry_crop",
+                     "L151.ag_irrProd_t_ctry_crop",
+                     "L151.ag_rfdProd_t_ctry_crop",
```

Comment:
Not sure whether these above should be called `precursors` or not. They are inputs for this chunk but not inputs for the final output.